src/routing_node.rs

```diff
@@ -65,19 +68,22 @@ static MAX_BOOTSTRAP_CONNECTIONS : usize = 1;
 /// Routing Node
 pub struct RoutingNode {
     // for CRUST
-    crust_receiver      : mpsc::Receiver<crust::Event>,
-    connection_manager  : crust::ConnectionManager,
-    accepting_on        : Vec<crust::Endpoint>,
-    bootstraps          : BTreeMap<Endpoint, Option<NameType>>,
+    crust_receiver        : mpsc::Receiver<crust::Event>,
+    connection_manager    : crust::ConnectionManager,
+    accepting_on          : Vec<crust::Endpoint>,
+    bootstraps            : BTreeMap<Endpoint, Option<NameType>>,
     // for RoutingNode
-    action_sender       : mpsc::Sender<Action>,
-    action_receiver     : mpsc::Receiver<Action>,
-    event_sender        : mpsc::Sender<Event>,
-    filter              : MessageFilter<types::FilterType>,
-    core                : RoutingCore,
-    public_id_cache     : LruCache<NameType, PublicId>,
-    connection_cache    : BTreeMap<NameType, SteadyTime>,
-    accumulator         : MessageAccumulator,
+    action_sender         : mpsc::Sender<Action>,
+    action_receiver       : mpsc::Receiver<Action>,
+    event_sender          : mpsc::Sender<Event>,
+    filter                : MessageFilter<types::FilterType>,
+    core                  : RoutingCore,
+    public_id_cache       : LruCache<NameType, PublicId>,
+    connection_cache      : BTreeMap<NameType, SteadyTime>,
+    accumulator           : MessageAccumulator,
+    immutable_data_cache  : LruCache<NameType, ImmutableData>,
```

Comment:
These should all be `Option` We will provide setters to any lib user who wishes to set size or time. It's up to users to set these. As this code stands we are setting them regardless. I don't think this is how we wanted it.